elastic/kibana
# [Dashboard Navigation] Fix reference extract method

Closes https://github.com/elastic/kibana/issues/164212
Closes https://github.com/elastic/kibana/issues/171328

## Summary

In `extractReferences`, the links in the `links` array attribute is [not extensible](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Cant_define_property_object_not_extensible) - this means that we cannot add or delete properties from each individual link directly. Instead, I've had to create a clone of each dashboard link via the spread operator so that we aren't modifying the original link directly. 

**How to Test**
1. Create a by-value Links panel
2. Refresh the dashboard
3. Save the Links panel to the library via the panel action
4. The panel should be saved successfully **and** the panel title should be updated.

[**Flaky Test Runner**](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/4024)

![image](https://github.com/elastic/kibana/assets/8698078/b8949d44-697d-414a-9e98-14b0a6f67b31)

### Checklist


### For maintainers
